packages/vscode-extension/src/commonlib/telemetry.ts

```diff
@@ -156,11 +195,30 @@ export class VSCodeTelemetryReporter extends vscode.Disposable implements Teleme
     }
 
     if (
-      properties[TelemetryProperty.ProjectId] === "unknown" ||
-      properties[TelemetryProperty.ProjectId] === undefined
+      !properties[TelemetryProperty.ProjectId] ||
```

Comment:
Maybe add a function `addFixedCommonProperties(properties, sharedProperties)` to prevent copy-pasting these code.

Reply:
Updated, thx!